extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/PreInitRecorder.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.amazon.lambda.runtime;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+/**
+ * The class pre-initialization recorder.
+ */
+@Recorder
+public final class PreInitRecorder {
+    public PreInitRecorder() {
+    }
+
+    public void preInit() {
+        final InputStream is = PreInitRecorder.class.getResourceAsStream("/META-INF/pre-init-classes.txt");
+        if (is != null)
+            try (is) {
+                try (BufferedInputStream bis = new BufferedInputStream(is)) {
+                    try (InputStreamReader isr = new InputStreamReader(bis, StandardCharsets.UTF_8)) {
+                        try (BufferedReader br = new BufferedReader(isr)) {
```

Comment:
I've always wondered: Is this the preferred way of using try-with-resources or should they all be in the same block? I vaguely remember Heinz Kabutz mentioning that the second pattern should be used, but I might be totally misremembering

Reply:
I always prefer this way; I find it more readable. But others might have different preferences of course.

Reply:
FWIW I don't like multiple try statements because it makes lines longer...For somebody who needs reading glasses, my fonts are huge and lines that must be scrolled in my editor makes me unhappy!  :)